// traveltime-api-sdk-ts/dist/auth/auth.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.configureAuthMethods = exports.ApplicationIdAuthentication = exports.ApiKeyAuthentication = void 0;
var ApiKeyAuthentication = (function () {
    function ApiKeyAuthentication(apiKey) {
        this.apiKey = apiKey;
    }
    ApiKeyAuthentication.prototype.getName = function () {
        return "ApiKey";
    };
    ApiKeyAuthentication.prototype.applySecurityAuthentication = function (context) {
        context.setHeaderParam("X-Api-Key", this.apiKey);
    };
    return ApiKeyAuthentication;
}());
exports.ApiKeyAuthentication = ApiKeyAuthentication;
var ApplicationIdAuthentication = (function () {
    function ApplicationIdAuthentication(apiKey) {
        this.apiKey = apiKey;
    }
    ApplicationIdAuthentication.prototype.getName = function () {
        return "ApplicationId";
    };
    ApplicationIdAuthentication.prototype.applySecurityAuthentication = function (context) {
        context.setHeaderParam("X-Application-Id", this.apiKey);
    };
    return ApplicationIdAuthentication;
}());
exports.ApplicationIdAuthentication = ApplicationIdAuthentication;
function configureAuthMethods(config) {
    var authMethods = {};
    if (!config) {
        return authMethods;
    }
    if (config["ApiKey"]) {
        authMethods["ApiKey"] = new ApiKeyAuthentication(config["ApiKey"]);
    }
    if (config["ApplicationId"]) {
        authMethods["ApplicationId"] = new ApplicationIdAuthentication(config["ApplicationId"]);
    }
    return authMethods;
}
exports.configureAuthMethods = configureAuthMethods;
